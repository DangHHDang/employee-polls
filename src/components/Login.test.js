import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import reducer from "../reducers";
import LogIn from "./Login";
const mockStore = configureStore([thunk]);

describe("LogIn", () => {
    let store, component;

    beforeEach(() => {
        store = mockStore({
            authedUser: null,
            users: {
                mtsamis: {
                    id: "mtsamis",
                    name: "Mike Tsamis",
                    password: "123456789"
                },
                zoshikanlu: {
                    id: "zoshikanlu",
                    name: "Zenobia Oshikanlu",
                    password: "123456789"
                },
            },
        });

        store.replaceReducer(reducer);

        component = render(
            <Provider store={store}>
                <Router>
                    <LogIn />
                </Router>
            </Provider>
        );
    });


    it("should match the snapshot", () => {
        expect(component).toMatchSnapshot();
    });
});