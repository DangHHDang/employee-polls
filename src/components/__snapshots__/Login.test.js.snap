// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="has-text-centered"
      >
        <div
          class="columns is-centered"
        >
          <div
            class="column is-two-fifths"
          >
            <h2
              class="is-size-2 my-5"
            >
              Login
            </h2>
            <form
              class="my-5"
            >
              <br />
              <div
                class="inputContainer"
              >
                <label>
                  Username
                  <input
                    class="inputBox"
                    data-testid="loginForm-username"
                    name="username"
                    placeholder="Enter your User Name here"
                    value=""
                  />
                </label>
                <p
                  class="errorLabel"
                />
              </div>
              <br />
              <div
                class="inputContainer"
              >
                <label>
                  Password
                  <input
                    class="inputBox"
                    data-testid="loginForm-password"
                    name="password"
                    placeholder="Enter your password here"
                    type="password"
                    value=""
                  />
                </label>
              </div>
              <br />
              <button
                class="button is-primary"
                name="log_in"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="has-text-centered"
    >
      <div
        class="columns is-centered"
      >
        <div
          class="column is-two-fifths"
        >
          <h2
            class="is-size-2 my-5"
          >
            Login
          </h2>
          <form
            class="my-5"
          >
            <br />
            <div
              class="inputContainer"
            >
              <label>
                Username
                <input
                  class="inputBox"
                  data-testid="loginForm-username"
                  name="username"
                  placeholder="Enter your User Name here"
                  value=""
                />
              </label>
              <p
                class="errorLabel"
              />
            </div>
            <br />
            <div
              class="inputContainer"
            >
              <label>
                Password
                <input
                  class="inputBox"
                  data-testid="loginForm-password"
                  name="password"
                  placeholder="Enter your password here"
                  type="password"
                  value=""
                />
              </label>
            </div>
            <br />
            <button
              class="button is-primary"
              name="log_in"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
